MobileBuy/buy/src/main/java/com/shopify/buy/dataprovider/CheckoutServiceDefault.java

```diff
@@ -153,12 +160,21 @@ public CancellableTask getShippingRates(final String checkoutToken, final Callba
             throw new NullPointerException("checkoutToken cannot be null");
         }
 
+        int[] successCodes = {HTTP_OK};
+
         return retrofitService
                 .getShippingRates(checkoutToken)
                 .retryWhen(networkRetryPolicyProvider.provide())
-                .doOnNext(new RetrofitSuccessHttpStatusCodeHandler<>())
+                .doOnNext(new RetrofitSuccessHttpStatusCodeHandler<>(successCodes))
+                .retryWhen(pollingRetryPolicyProvider.provide())
                 .compose(new UnwrapRetrofitBodyTransformer<ShippingRatesWrapper, List<ShippingRate>>())
-                .observeOn(callbackScheduler);
+                .observeOn(callbackScheduler)
+                .doOnError(new Action1<Throwable>() {
+                    @Override
+                    public void call(Throwable throwable) {
+                        System.out.println(throwable);
```

Comment:
We need to introduce SDK logger, as this is not the only place where it's required